### PR TITLE
[ref arch] updates for GCP & Team Workspace

### PR DESCRIPTION
**Pull request by kylos101**
This adjusts the Kubernetes cluster so that it isolates workspaces from services, isolates headless workspaces from regular workspaces, and shares a single cluster ref arch is not highly available.
 
- [x] add headless node pool
- [x] move workspace services to the services pool
- [x] add high availability limitation

For awareness, the Terraform modules @gitpod-io/engineering-self-hosted will need to be updated accordingly.

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2298"></a>

I'll create a separate PR (which'll require discussion and performance testing) to focus on:
-  Socializing workspaces per node versus headless workspaces per node (headless are resource hungry, lower density helps)
-  Socialize storage requires (in general) for each workspace (which impacts the disk size for nodes)
-  Disk number and size (which impacts perf, at least for GCP)
-  Inspecting default Kubernetes requests and limits for workspaces, and adjusting accordingly
-  Inspecting Gitpod `ws-daemon` CPU limiting, and adjusting accordingly
-  Updating the ReferenceArchitecture picture, to show key missing services (`ws-manager` to services, and `blobserve` to ide), and move existing ones (`registry-facade`) to workspace and headless

### CODE REVIEW

**Comment by kylos101** [comment]
@corneliusludmann :wave: hey there, I've made some changes for Team Workspace. I've got one more thing to think about (recommendations for density), but, wanted to give you a heads up to get your thoughts. LMKWYT?

**Comment by lucasvaltl** [comment]
Hey @kylos101 were you able to think more about recommendations for density? I'll also ping @corneliusludmann to tal at this :) From my side: thanks for updating the prod compatibility matrix version numbers. Super helpful!

**Comment by lucasvaltl** [comment]
Note that when we merge this, we'll also need to update the corresponding terraform configuration here: https://github.com/gitpod-io/gitpod/tree/main/install/infra/single-cluster

**Comment by kylos101** [comment]
> Note that when we merge this, we'll also need to update the corresponding terraform configuration here: https://github.com/gitpod-io/gitpod/tree/main/install/infra/single-cluster

@lucasvaltl there will probably be a few changes needed in Terraform. Can that be done in a follow-on PR? I ask because I lack related experience, which'll cause this update to be delayed.

File: gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md [hunk 1]
```diff
@@ -6,24 +6,27 @@ layout: false
...
 The following table gives an overview of the node types for the different cloud providers that are used by this reference architecture.
 
-|                      | GCP               | AWS           |
-| -------------------- | ----------------- | ------------- |
-| Services Node Pool   | `n2d-standard-8`  | `m6i.xlarge`  |
-| Workspaces Node Pool | `n2d-standard-16` | `m6i.2xlarge` |
+|                              | GCP               | AWS           |
+| ---------------------------- | ----------------- | ------------- |
+| Services Node Pool           | `n2d-standard-8`  | `m6i.xlarge`  |
+| Regular Workspace Node Pool  | `n2d-standard-16` | `m6i.2xlarge` |
```
[the remark this answers is not in the document]
Comment (kylos101) [reply gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md:28]: Okay, good to know, thanks!

**Comment by lucasvaltl** [comment]
> @lucasvaltl there will probably be a few changes needed in Terraform. Can that be done in a follow-on PR? I ask because I lack related experience, which'll cause this update to be delayed.

Yes :) Just have to coordinate the releases of the respective PRs somewhat that the difference in guide vs. terraform does not exist for too long. The self-hosted team is happy to assist with the terraform changes :)

File: gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylos101) [reply gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md:28]: :wave: @lucasvaltl can you help me understand context for [this](https://github.com/gitpod-io/website/pull/2298#discussion_r954793212)? 

Some questions I have:
1. Why did we downsize? Can you share a link to a related conversation?
2. How many workspaces are we targeting to run with this configuration? Is there a range we're targeting? Was there any discussion about what to do if the range was exceeded?
3. When I look at [main](https://github.com/gitpod-io/website/blob/main/gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md), the node types I see do not end with `s`. Why the change to node type ending with `s`? Is the `s` suffix you added a typo, or a reference to a branch that hasn't been merged yet?

[the remark this answers is not in the document]
Comment (kylos101) [reply gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md:28]: I see [this](https://www.gitpod.io/docs/self-hosted/latest/reference-architecture/single-cluster-ref-arch#:~:text=*%20With%20the%20configuration%20in%20these%20estimates%2C%20you%20are%20able%20to%20run%2036%20concurrent%20basic%20workspaces%20%2D%20assuming%20the%20same%20density%20as%20currently%20used%20in%20Gitpod%20SaaS.):
`* With the configuration in these estimates, you are able to run 36 concurrent basic workspaces - assuming the same density as currently used in Gitpod SaaS. `

Which indicates the plan is to run 12 workspaces per node. That seems high considering:
1. The downsized nodes are half of what we use in prod
2. In prod, we run ~18-19 workspaces per node

[the remark this answers is not in the document]
Comment (kylos101) [reply gitpod/docs/self-hosted/latest/reference-architecture/_chunks/cluster.md:28]: Given [this](https://cloud.google.com/compute/docs/general-purpose-machines#n2d-standard), I'm going to assume you meant something like:

> n2d-standard-8's for workspaces, and n2d-standard-4's for services

Lemme know on the other questions?

**Comment by kylos101** [comment]
Closing in favor of https://github.com/gitpod-io/website/pull/2652

**Pull request closed without merging**